### PR TITLE
Clean-up of PathUtil 

### PR DESCRIPTION
**Pull request by edgarmueller**
Small refactoring of PathUtil + JSDoc, also fixed some minor warnings.
Usage of [start case](https://en.wikipedia.org/wiki/Letter_case#Stylistic_or_specialised_usage) when beautifying labels.
Fixed name of inits function, which actually should be named init (as e.g. [here](http://www.scala-lang.org/api/2.7.7/scala/List.html#init)).

### CODE REVIEW

**Comment by edgarmueller** [comment]
@eneufeld This is a refactoring I did in my spare time to explore lodash. No issue or whatsoever tracks this.

File: src/components/services/pathutil.ts [hunk 1]
```diff
@@ -1,80 +1,110 @@
...
     }
 
+    /**
+     * Returns the last fragment of the given path in a beautified manner.
+     *
+     * @param schemaPath the path from which to extract the beautified fragment
+     * @returns {any} the last fragment in a beautified manner
+     */
+    static beautifiedLastFragment(schemaPath: string): string  {
+        return _.flow(PathUtil.lastFragment, _.upperFirst, PathUtil.beautify)(schemaPath);
```
Comment (eneufeld) [inline src/components/services/pathutil.ts]: the _.uperFirst call is not necessary as the beautify method will upper case the string

File: src/components/services/pathutil.spec.ts [hunk 2]
```diff
@@ -3,12 +3,28 @@ import "angular-mocks"
...
+        expect(PathUtil.lastFragment("foo/bar")).toBe("bar")
+    });
+
+    it("should beautify the last fragment", () => {
+        expect(PathUtil.beautifiedLastFragment("foo/bar")).toBe("Bar")
+    });
+
+    it("should filter indices from a path", () => {
+        expect(PathUtil.filterIndexes("foo/items/1/bar")).toBe("foo/items/bar")
+    });
```
Comment (eneufeld) [inline src/components/services/pathutil.spec.ts]: a test case for a beautify of a property like firstName to be "First Name" would be good

**Comment by eneufeld** [comment]
Thank you, the code is much better readable. I added two small comments, but they should/could be fixed separately as they are improvements

**Pull request merged**